src/Microsoft.IdentityModel.Clients.ActiveDirectory/Internal/Platform/LoggerBase.cs

```diff
@@ -47,9 +60,49 @@ internal static string GetCallerFilename(string callerFilePath)
 
         internal string CorrelationId { get; set; } = string.Empty;
 
-        internal string PrepareLogMessage(CallState callState, string classOrComponent, string message)
+        internal string FormatLogMessage(CallState callState, string classOrComponent, string message)
         {
             return string.Format(CultureInfo.InvariantCulture, "{0:O}: {1} - {2}: {3}", DateTime.UtcNow, CorrelationId, classOrComponent, message);
         }
+
+        internal void Verbose(CallState callState, string message)
+        {
+            Log(callState, LogLevel.Verbose, message, false);
+        }
+
+        internal void VerbosePii(CallState callState, string message)
+        {
+            Log(callState, LogLevel.Verbose, message, true);
+        }
+
+        internal void Information(CallState callState, string message)
+        {
+            Log(callState, LogLevel.Information, message, false);
+        }
+
+        internal void InformationPii(CallState callState, string message)
+        {
+            Log(callState, LogLevel.Information, message, true);
+        }
+
+        internal void Warning(CallState callState, string message)
+        {
+            Log(callState, LogLevel.Warning, message, false);
+        }
+
+        internal void WarningPii(CallState callState, string message)
+        {
+            Log(callState, LogLevel.Warning, message, true);
+        }
+
+        internal void Error(CallState callState, Exception ex)
+        {
+            Log(callState, LogLevel.Error, ex.ToString(), false);
```

Comment:
ex.ToString may have PII, it is hard to tell. Should we use ex.GetType() when PII is off.

Reply:
I feel like ex.GetType()  is not quite informative, may be even almost useless for troubleshooting,
taking to account that PII logging is disabled by default It might be a problem